pkg/activitypub/store/memstore/memstore.go

```diff
@@ -209,36 +213,80 @@ func (s *referenceStore) get(actor fmt.Stringer) ([]*url.URL, error) {
 	return s.irisByActor[actor.String()], nil
 }
 
-type queryFilter struct {
+type activityQueryFilter struct {
 	*spi.Criteria
 }
 
-func newQueryFilter(query *spi.Criteria) *queryFilter {
-	return &queryFilter{
+func newQueryFilter(query *spi.Criteria) *activityQueryFilter {
+	return &activityQueryFilter{
 		Criteria: query,
 	}
 }
 
-func (q *queryFilter) Accept(activity *vocab.ActivityType) bool {
-	if len(q.Types) > 0 {
-		return activity.Type().IsAny(q.Types...)
+func (q *activityQueryFilter) apply(activities []*vocab.ActivityType) []*vocab.ActivityType {
+	var results []*vocab.ActivityType
+
+	for _, a := range activities {
+		if len(q.Types) == 0 || a.Type().IsAny(q.Types...) {
+			results = append(results, a)
+		}
 	}
 
-	return true
+	return results
 }
 
-type memQueryResults map[string]*vocab.ActivityType
+type activityQueryResults []*vocab.ActivityType
 
-func (r memQueryResults) filter(query *spi.Criteria) []*vocab.ActivityType {
-	var results []*vocab.ActivityType
+func (r activityQueryResults) filter(query *spi.Criteria, opts ...spi.QueryOpt) ([]*vocab.ActivityType, int) {
```

Comment:
For consistency, you may want to make this a pointer receiver

Reply:
activityQueryResults is actually a slice (not a struct)

Reply:
Oh, I see. Please disregard my comment! :)